Tests/SwiftDiagnosticsTest/GroupDiagnosticsFormatterTests.swift

```diff
@@ -35,34 +25,22 @@ extension GroupedDiagnostics {
     _ markedSource: String,
     displayName: String,
     parent: (SourceFileID, AbsolutePosition)? = nil,
-    extraDiagnostics: [String: (String, DiagnosticSeverity)] = [:]
+    diagnosticDescriptors: [DiagnosticDescriptor]
   ) -> (SourceFileID, [String: AbsolutePosition]) {
-    // Parse the source file and produce parser diagnostics.
     let (markers, source) = extractMarkers(markedSource)
     let tree = Parser.parse(source: source)
-    var diagnostics = ParseDiagnosticsGenerator.diagnostics(for: tree)
-
-    // Add on any extra diagnostics provided, at their marker locations.
-    for (marker, (message, severity)) in extraDiagnostics {
-      let pos = AbsolutePosition(utf8Offset: markers[marker]!)
-      let node = tree.token(at: pos)!.parent!
-
-      let diag = Diagnostic(
-        node: node,
-        message: SimpleDiagnosticMessage(
-          message: message,
-          diagnosticID: MessageID(domain: "test", id: "conjured"),
-          severity: severity
-        )
-      )
-      diagnostics.append(diag)
+
+    let parserDiagnostics = ParseDiagnosticsGenerator.diagnostics(for: tree)
```

Comment:
Regarding my comment above about why we need to check that the tree didn’t contain any errors, I think if we removed this line as well and only displayed errors based on `diagnosticDescriptors`, we would be completely independent of `SwiftParserDiagnostics` here.

Reply:
I love the idea, but I'd prefer to hold off on that for this PR. `GroupedDiagnostics` has its own set of issues, and if we were to remove this part now, we'd likely find ourselves fixing additional problems. After I've improved `DiagnosticsFormatter`, I plan on revisiting `GroupedDiagnostics`. I'll definitely tackle it at that point.

Reply:
OK sounds good to me.